### ISSUE TITLE
 Module '"../@types/domhandler"' has no exported member

### ISSUE THREAD

**Issue by fabienbranchel** [remarkablemark/html-react-parser#231]
## Expected Behavior

Run OK

## Actual Behavior

Run KO

## Steps to Reproduce

When used with TypeScript (4.1.5), it's suggest to install `@types/domhandler` and `@types/htmlparser2`.

It's fine. But when it's done, I've got these errors :

![image](https://user-images.githubusercontent.com/23743473/108999192-420f2380-76a2-11eb-8545-e8f699f644c4.png)

## Environment

- Version: 1.2.4
- Platform: macOS
- Browser: Chrome

**Comment by remarkablemark** [comment]
@fabienbranchel can you import the types without using `@types`?

```ts
import { Comment, ProcessingInstruction, Text } from 'html-react-parser';
```

**Comment by fabienbranchel** [comment]
I try, it's not resolving the issue.
Moreover, it's doesn't feel right to import something I don't use directly.

**Comment by remarkablemark** [comment]
So [domhandler](https://github.com/fb55/domhandler/blob/v4.0.0/package.json#L11) exports its own types so installing `@types/domhandler` may be unnecessary.

Can you provide a reproducible example using [CodeSandbox](https://codesandbox.io/s/html-react-parser-z0kp6)?

**Comment by kfritsch** [comment]
### Fix 
Hey I got the same Issue. I think the fix would be adding domhandler ^4.0.0 as a dependency.

### Explanation
The problem seems to be that this package uses "domhandler" directly but does not have it as a direct dependency. It only has it as an indirect dependency via "html-dom-parser". This can cause problems with the way npm installs packages. If another package requires an older version of "domhandler" and is installed first it will be placed at the root level of node_modules. Since this package does not have an explicit dependency on "domhandler" it will use what is installed at the root level. That "html-dom-parser" uses the required newer version does not help since this version is now placed inside "html-dom-parser".

- /node_modules
  - /domhandler (v3.0.0)
  - /html-dom-parser
    - /domhandler (v.4.0.0)
  - /html-react-parser

### Workaround 
The installation of the old "domhandler" 3.0.0 is actually triggered by the dependencies of this module. This means that the only other workaround would be adding "domhandler" ^4.0.0 before "html-react-parser" in the dependencies so it ends up at the root level.

**Comment by remarkablemark** [comment]
@kfritsch Thanks for opening the PR. I left some comments.

**Comment by remarkablemark** [comment]
@fabienbranchel @kfritsch I'll make a release later tonight

**Comment by remarkablemark** [comment]
Published [v1.2.5](https://github.com/remarkablemark/html-react-parser/releases/tag/v1.2.5):

[npm](https://www.npmjs.com/package/html-react-parser):

```sh
npm i html-react-parser@1.2.5
```

[Yarn](https://yarnpkg.com/package/html-react-parser):

```sh
yarn add html-react-parser@1.2.5
```
